tests/metatype/tmetatype_issues.nim

```diff
@@ -156,3 +156,11 @@ block t3338:
   var t2 = Bar[int32]()
   t2.add()
   doAssert t2.x == 5
+
+block t9961:
+  proc n(t:typedesc) =
+    echo t.type.name
+  const t : typedesc = int
+  assert(not
+         compiles((block:
```

Comment:
/cc @araq 
that'd be a perfect, typical use case for `compilesWithContext` I suggested here: https://github.com/nim-lang/Nim/issues/8803
currently the error message isn't being tested for, and PR author @deech would have to run through significant contorsions to test for it (moving this to a separate test file for example). I run into this all the time. I'm willing to try implementing it if feature is pre-approved.


here, its use would look like:

```nim
let context = compilesWithContext(block: (n t))
doAssert: not context.ok
doAssert: context. errorMsg.startsWith("Could not find a matching ...")
```

Reply:
Again, error messages can be checked for ``cmd: "nim check $file"``. Again, ``compilesWithContext`` is the worst hack in the history of mankind and won't be accepted.